onnxruntime/core/providers/dml/DmlExecutionProvider/src/Operators/DmlDFT.h

```diff
@@ -0,0 +1,695 @@
+
+        // Update the pending element count
+        pendingElementCount = (dispatchedElementCount < elementCount) ? elementCount - dispatchedElementCount : 0;
+    }
+
+}
+
+class GpuDFTOperator : public WRL::Base<IMLOperatorKernel>
+{
+private:
+    ComPtr<ID3D12Device> m_device;
+    ComPtr<ID3D12RootSignature> m_rootSignature;
+    ComPtr<ID3D12PipelineState> m_pipelineState;
+
+    std::vector<uint32_t> m_inputDims = {};
+    std::vector<uint32_t> m_outputDims = {};
+    int64_t m_axis;
+    bool m_isOnesided;
+    bool m_isInverse;
+
+    uint32_t m_outputDataSize = 0;
+    uint32_t m_inputDataSize = 0;
+    uint32_t m_outputIdx = 0;
+    uint32_t m_numPasses = 0;
+
+    // Allocate temporary buffers if needed
+    struct ResourceDesc
+    {
+        ComPtr<ID3D12Resource> Resource;
+        std::array<uint32_t, 4> Sizes;
+        std::array<uint32_t, 4> Strides;
+    };
+    std::vector<ResourceDesc> resourceLoopList_ = {};
+
+    struct LoopRange
+    {
+        unsigned Left;
+        unsigned Right;
+        unsigned End;
+        unsigned CalculateIndex(unsigned index) {
+            if (index > 0 && index < End)
+            {
+                unsigned range = Right - Left + 1;
+                index = Left + (index - 1) % range;
+            }
+            else if (index == End) {
+                index = Right + 1;
+            }
+            return index;
+        }
+    };
+    LoopRange loopRange_ = {};
+
+    struct DFTShaderConstants
+    {
+        uint32_t StartIndex;
+        uint32_t ElementCount;
+        uint32_t DFTIteration;
+        uint32_t IsInverse;
+        uint32_t InputSizes[4];
+        uint32_t InputStrides[4];
+        uint32_t OutputSizes[4];
+        uint32_t OutputStrides[4];
+        float Scale;
+    };
+
+public:
+    GpuDFTOperator(IMLOperatorKernelCreationContext* context)
+    {
+        ComPtr<IUnknown> executionObject;
+        context->GetExecutionInterface(executionObject.GetAddressOf());
+
+        ComPtr<ID3D12GraphicsCommandList> commandList;
+        executionObject.As(&commandList);
+
+        ORT_THROW_IF_FAILED(commandList->GetDevice(IID_ID3D12Device, &m_device));
+
+
+        ORT_THROW_IF_FAILED(context->GetAttribute("axis", MLOperatorAttributeType::Int, 1, sizeof(int64_t), reinterpret_cast<void*>(&m_axis)));
+
+        int64_t isInverseInt;
+        ORT_THROW_IF_FAILED(context->GetAttribute("inverse", MLOperatorAttributeType::Int, 1, sizeof(int64_t), reinterpret_cast<void*>(&isInverseInt)));
+        m_isInverse = static_cast<bool>(isInverseInt);
+
+        int64_t isOnesidedInt;
+        ORT_THROW_IF_FAILED(context->GetAttribute("onesided", MLOperatorAttributeType::Int, 1, sizeof(int64_t), reinterpret_cast<void*>(&isOnesidedInt)));
+        m_isOnesided = static_cast<bool>(isOnesidedInt);
+
+        ComPtr<IMLOperatorTensorShapeDescription> shapeDesc;
+        context->GetTensorShapeDescription(shapeDesc.GetAddressOf());
```

Comment:
>context->GetTensorShapeDescription(shapeDesc.GetAddressOf()); [](http://example.com/codeflow?start=8&length=61)

ORT_THROW_IF_FAILED?

Since this method can fail too:

```
          if (!HasTensorShapeDescription())
          {
            *shapeInfo = nullptr;
            return E_FAIL;
            //return MLStatus::REQUIREMENT_NOT_REGISTERED;
          }
```